libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationEngineFactory.kt

```diff
@@ -0,0 +1,73 @@
+package com.mapbox.services.android.navigation.v5.internal.navigation
+
+import com.mapbox.services.android.navigation.v5.navigation.camera.Camera
+import com.mapbox.services.android.navigation.v5.navigation.camera.SimpleCamera
+import com.mapbox.services.android.navigation.v5.offroute.OffRoute
+import com.mapbox.services.android.navigation.v5.offroute.OffRouteDetector
+import com.mapbox.services.android.navigation.v5.route.FasterRoute
+import com.mapbox.services.android.navigation.v5.route.FasterRouteDetector
+import com.mapbox.services.android.navigation.v5.snap.Snap
+import com.mapbox.services.android.navigation.v5.snap.SnapToRoute
+
+class NavigationEngineFactory {
+
+    private var offRouteEngine: OffRoute? = null
+    private var fasterRouteEngine: FasterRoute? = null
+    private var snapEngine: Snap? = null
+    private var cameraEngine: Camera? = null
+
+    init {
+        initializeDefaultEngines()
+    }
+
+    fun retrieveOffRouteEngine(): OffRoute? {
+        return offRouteEngine
+    }
+
+    fun updateOffRouteEngine(offRouteEngine: OffRoute?) {
+        if (offRouteEngine == null) {
```

Comment:
Could we remove `null` checks if we use non-nullable types (see comment above)?

Reply:
👍